Dropdown/DropdownList.js

```diff
@@ -0,0 +1,313 @@
+					cbScrollTo={scrollTo}
+					className={css.virtualList}
+					dataSize={dataSize}
+					itemRenderer={itemRenderer}
+					itemSize={itemSize}
+				/>
+			</div>
+		);
+	}
+});
+
+const ReadyState = {
+	// Initial state. Scrolling and focusing pending
+	INIT: 0,
+	// Scroll requested
+	SCROLLED: 1,
+	// Focus completed or not required
+	DONE: 2
+};
+
+const DropdownListSpotlightDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'DropdownListSpotlightDecorator';
+
+		static propTypes = {
+			/**
+			 * Called when an item receives focus.
+			 *
+			 * @type {Function}
+			 */
+			onFocus: PropTypes.func,
+
+			/**
+			 * Index of the selected item.
+			 *
+			 * @type {Number}
+			 */
+			selected: PropTypes.number
+		};
+
+		constructor (props) {
+			super(props);
+
+			this.state = {
+				prevChildren: props.children,
+				prevFocused: null,
+				prevSelected: this.props.selected,
+				prevSelectedKey: getKey(props),
+				ready: isSelectedValid(props) ? ReadyState.INIT : ReadyState.DONE
+			};
+		}
+
+		componentDidMount () {
+			// eslint-disable-next-line react/no-find-dom-node
+			this.node = ReactDOM.findDOMNode(this);
+			Spotlight.set(this.node.dataset.spotlightId, {
+				defaultElement: '[data-selected="true"]',
+				enterTo: 'default-element'
+			});
+		}
+
+		componentDidUpdate () {
+			if (this.state.ready === ReadyState.INIT) {
+				this.scrollIntoView();
+			} else if (this.state.ready === ReadyState.SCROLLED) {
+				this.focusSelected();
+			} else {
+				const key = getKey(this.props);
+				const keysDiffer = key && this.state.prevSelectedKey && key !== this.state.prevSelectedKey;
+
+				if (keysDiffer ||
+					((!key || !this.state.prevSelectedKey) && this.state.prevSelected !== this.props.selected) ||
+					!compareChildren(this.state.prevChildren, this.props.children)
+				) {
+					this.resetFocus(keysDiffer);
+				}
+			}
+		}
+
+		setScrollTo = (scrollTo) => {
+			this.scrollTo = scrollTo;
+		};
+
+		resetFocus (keysDiffer) {
+			let adjustedFocusIndex;
+
+			if (!keysDiffer && this.lastFocusedKey) {
+				const targetIndex = indexFromKey(this.props.children, this.lastFocusedKey);
+				if (targetIndex >= 0) {
+					adjustedFocusIndex = targetIndex;
+				}
+			}
+
+			this.setState({
+				prevChildren: this.props.children,
+				prevFocused: adjustedFocusIndex,
+				prevSelected: this.props.selected,
+				prevSelectedKey: getKey(this.props),
+				ready: ReadyState.INIT
+			});
+		}
+
+		scrollIntoView = () => {
+			let {selected} = this.props;
+
+			if (this.state.prevFocused == null && !isSelectedValid(this.props)) {
+				selected = 0;
+			} else if (this.state.prevFocused != null) {
+				selected = this.state.prevFocused;
+			}
+
+			this.scrollTo({
+				animate: false,
+				focus: true,
+				index: selected,
+				offset: ri.scale(126 * 2), // @sand-item-small-height * 2 (TODO: large text mode not supported!)
+				stickTo: 'start' // offset from the top of the dropdown
```

Comment:
These were for centering the focused item when the dropdown list is opened in sandstone.
If we don't need this, we should remove it. I think because of these options, the selected item is not shown when it's opened in silicon skin like below.
![image](https://user-images.githubusercontent.com/5037429/108972751-f6bb3c00-76c6-11eb-802f-aec399d7d1c2.png)